UnityProject/Assets/Scripts/HealthV2/Living/LivingHealthMasterBase.cs

```diff
@@ -493,17 +493,21 @@ public void CalculateOverallHealth()
 				bool hasAllHeartAttack = true;
 				foreach (var Implant in ImplantList)
 				{
-					var heart = Implant as Heart;
-					if (heart != null)
+					foreach (var bodyPartModification in Implant.BodyPartModifications)
 					{
-						if (heart.HeartAttack == false)
+						var heart = bodyPartModification as Heart;
+						if (heart != null)
```

Comment:
wouldn't ``if (heart is Heart)`` work for this?

Reply:
in the live above, instead of getting as heart ``if (bodyPartModification is Heart)``

Reply:
well you need to access it straight after so it's more efficient to
 cast then check then ues
cast then cast and then use

Reply:
according to redline, using ``is`` will cast + null check + assign to the variable.

Reply:
idk how it assigns